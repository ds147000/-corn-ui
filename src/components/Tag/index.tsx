import React, { useMemo } from 'react'
import { View, Text } from '@tarojs/components'
import ClassNames from 'classnames'
import type { ViewProps } from '@tarojs/components/types/View'

export type TagSize = 'middle' | 'small'
export type TagType =  'primary' | 'warn' | 'error' | 'link' | 'pop'

export interface TagProps extends ViewProps {
  size?: TagSize
  /** 按钮风格类型 */
  type?: TagType
  /** 镂空风格 */
  ghost?: boolean
}

const Tag: React.FC<TagProps> = ({ size = 'small', type = 'primary', ghost, children, className, ...props }) => {
  const _class = useMemo(() => {
    return ClassNames(
      'xrk-tag xrk-if xrk-ac',
      'xrk-tag-' + size,
      'xrk-tag-' + type,
      ghost && 'xrk-tag-ghost',
      className
    )
  }, [ className, size, type, ghost ])
  return (
    <View className={_class} {...props}>
      <Text>{children}</Text>
    </View>
  )
}

export default Tag
