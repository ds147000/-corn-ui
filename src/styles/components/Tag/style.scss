.xrk-tag {
  box-sizing: border-box;
  white-space: nowrap;

  &-large {
    height: 46px;
    padding: 0 10px;
    font-size: 24px;
    font-weight: bold;
    line-height: normal;
    margin: 0 8px;
    border-radius: 8px;
    border: 2px solid rgba($color: #000 , $alpha: 0);
  }

  &-middle {
    height: 42px;
    padding: 0 10px;
    font-size: 22px;
    font-weight: bold;
    line-height: normal;
    margin: 0 8px;
    border-radius: 8px;
    border: 2px solid rgba($color: #000 , $alpha: 0);
  }

  &-small {
    padding: 0 9px;
    border-radius: 6px;
    height: 30px;
    font-size: 21px;
    line-height: normal;
    font-weight: bolder;
    margin: 0 8px;
    border: 1px solid rgba($color: #000 , $alpha: 0);
  }

  &-mini {
    padding: 0 8px;
    border-radius: 6px;
    height: 26px;
    font-size: 18px;
    line-height: 18px;
    font-weight: bolder;
    margin: 0 6px;
    border: 1px solid rgba($color: #000 , $alpha: 0);
  }

  &-primary {
    background-color: $primary;
    color: $text1;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $primary , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $primary;
    }
  }

  &-warn {
    background-color: $warn;
    color: $error;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $warn , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $warn;
    }
  }

  &-error {
    background-color: $error;
    color: $white;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $error , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $error;
    }
  }

  &-link {
    background-color: $link;
    color: $white;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $link , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $link;
    }
  }

  &-pop {
    background-color: $pop;
    color: $white;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $pop , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $pop;
    }
  }

  &-defautl {
    background-color: $defualt;
    color: $white;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $defualt , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $defualt;
    }
  }

  &-urgent {
    background-color: $urgentBack;
    color: $urgent;

    &.xrk-tag-ghost {
      border: 1px solid rgba($color: $urgent , $alpha: 1);
      background-color: rgba($color: $white , $alpha: 0);
      color: $urgent;
    }
  }

  &-activity {
    background: $activity;
    color: $white;
    font-style: oblique;
  }
}
