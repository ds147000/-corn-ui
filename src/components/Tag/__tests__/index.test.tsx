import React from 'react'
import { render } from '@testing-library/react'
import Tag from '../index'

describe('Tag', () => {
  test('base', () => {
    const screen = render(<Tag>3-9岁</Tag>)
    expect(screen).toMatchSnapshot()
  })

  test('size', () => {
    const screen = render(
      <>
        <Tag size="large">3-9岁</Tag>
        <Tag size="middle">3-9岁</Tag>
        <Tag size="small">3-9岁</Tag>
        <Tag size="mini">3-9岁</Tag>
      </>
    )
    expect(screen).toMatchSnapshot()
  })

  test('type', () => {
    const screen = render(
      <>
        <Tag type="error">3-9岁</Tag>
        <Tag type="link">3-9岁</Tag>
        <Tag type="pop">3-9岁</Tag>
        <Tag type="primary">3-9岁</Tag>
        <Tag type="warn">3-9岁</Tag>
        <Tag type="urgent">3-9岁</Tag>
        <Tag type="activity">3-9岁</Tag>
      </>
    )
    expect(screen).toMatchSnapshot()
  })

  test('type', () => {
    const screen = render(
      <>
        <Tag type="error" ghost>3-9岁</Tag>
        <Tag type="link" ghost>3-9岁</Tag>
        <Tag type="pop" ghost>3-9岁</Tag>
        <Tag type="primary" ghost>3-9岁</Tag>
        <Tag type="warn" ghost>3-9岁</Tag>
        <Tag type="urgent" ghost >3-9岁</Tag>
      </>
    )
    expect(screen).toMatchSnapshot()
  })
})
