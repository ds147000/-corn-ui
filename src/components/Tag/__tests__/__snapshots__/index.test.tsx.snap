// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag base 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-large xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-middle xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-mini xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-large xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-middle xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-mini xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag type 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-error"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-link"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-pop"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-warn"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-urgent"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-activity"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-error"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-link"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-pop"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-warn"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-urgent"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-activity"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag type 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-error xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-link xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-pop xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-warn xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
      <div
        class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-urgent xrk-tag-ghost"
      >
        <span
          class="taro-text"
        >
          3-9岁
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-error xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-link xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-pop xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-primary xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-warn xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
    <div
      class="xrk-tag xrk-if xrk-ac xrk-tag-small xrk-tag-urgent xrk-tag-ghost"
    >
      <span
        class="taro-text"
      >
        3-9岁
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
